java/src/main/java/com/google/spsproject/servlets/User.java

```diff
@@ -0,0 +1,87 @@
+package com.google.spsproject.servlets;
+
+import com.google.spsproject.data.UserInfo;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse; 
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.api.core.ApiFuture;
+import java.util.*;
+
+
+@WebServlet("/user")
+public class User extends HttpServlet {
+    Firestore db;
+    public User() throws IOException {
+        FirestoreOptions firestoreOptions =
+        FirestoreOptions.getDefaultInstance().toBuilder()
+        .setProjectId("summer20-sps-77")
+        .setCredentials(GoogleCredentials.getApplicationDefault())
+        .build();
+        this.db = firestoreOptions.getService();
+    }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String email = request.getParameter("email");
+        DocumentReference docRef = db.collection("users").document(email);
+        ApiFuture<DocumentSnapshot> future = docRef.get();
+        UserInfo UserObject = new UserInfo();
+        try {
+            DocumentSnapshot document = future.get();
```

Comment:
Just FYI when you do future.get() it waits for the task to finish so it's not really async, not an issue but you should assume it is